esp8266/modesp.c

```diff
@@ -35,6 +35,492 @@
 #include MICROPY_HAL_H
 #include "queue.h"
 #include "user_interface.h"
+#include "espconn.h"
+
+#include "utils.h"
+
+#define IPV4ADDR_BUFSIZE    4
+
+// Helpers from cc3200/mods/modnetwork.c
+STATIC void esp_espconn_parse_ipv4_addr(mp_obj_t addr_in, uint8_t *out_ip) {
```

Comment:
It's not hard technically, only logistically - where should it be moved, how far you should go, how many ports will be broken and need fixing, etc. Thanks for trying it!